File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        unordered_set<int> set;
        vector<int> ans;
        int a, b;
        int actual_sum = 0;
        int n = grid.size();

        for(int i = 0; i<n; i++){
            for(int j = 0; j<n; j++){
                actual_sum += grid[i][j];
                if(set.find(grid[i][j]) != set.end()){
                    a = grid[i][j];
                    ans.push_back(a);
                }

                set.insert(grid[i][j]);
            }
        }

        int expected_sum = (n*n*(n*n + 1))/2;
        b = expected_sum  - actual_sum + a;
        ans.push_back(b);

        return ans;
    }
};
    

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n = grid.length;
        int nSquared = n * n;
        int[] count = new int[nSquared + 1];

        // Count occurrences of each number in the grid
        for (int[] row : grid) {
            for (int num : row) {
                count[num]++;
            }
        }

        int repeated = -1;
        int missing = -1;

        // Identify the repeated and missing numbers
        for (int i = 1; i <= nSquared; i++) {
            if (count[i] == 2) {
                repeated = i;
            } else if (count[i] == 0) {
                missing = i;
            }
        }

        return new int[]{repeated, missing};
    }
}